lib/manager/gradle/index.js

```diff
@@ -0,0 +1,160 @@
+const { exec } = require('child-process-promise');
+const fs = require('fs-extra');
+const path = require('path');
+
+const gradle = require('./build-gradle');
+
+const GRADLE_DEPENDENCY_REPORT_COMMAND =
+  'gradle --init-script init.gradle dependencyUpdates -Drevision=release';
+const GRADLE_DEPENDENCY_REPORT_FILENAME = 'build/dependencyUpdates/report.json';
+
+async function preExtract(config, packageFiles) {
```

Comment:
I had to create a preExtract (similar to the postExtract) method because before calling gradle I need to make sure all the gradle.build and settings.gradle files has been saved. 

Because the extractDependencies method is called for every file independently I don't have a way to know when all the files has been saved. So I need a place to save them first, and then call gradle.